bundles/org.openhab.binding.sensibo/src/main/java/org/openhab/binding/sensibo/internal/handler/SensiboSkyHandler.java

```diff
@@ -75,15 +73,14 @@
  *
  * @author Arne Seime - Initial contribution
  */
-@NonNullByDefault
```

Comment:
The class should be annotated with `@NonNullByDefault`, the config should be `@Nullable`. Unfortunately this requires null-checks wherever the config is used. I'm wondering if we should use

```
private Optional<SensiboSkyConfiguration> config = Optional.empty();
```

We could use `config.ifPresent(cfg -> ...);` then instead of a null check when we need to access the config. WDYT?

Reply:
It is going to be a lot of unnecessary boilerplate code (4 places) just to please the annotation processor. IMHO it will also reduce both maintainability and readability of the code.

Reply:
I'm not very happy with that, too. IMO `@NonNullByDefault({})` would  be a good fit. But since that is not injected (like an OSGI service) it was decided that this should not be used. So we either need `@Nullable` and null-checks where the config is used or the `Optional`.

Reply:
Using optional with helper method